localstack/services/route53resolver/provider.py

```diff
@@ -0,0 +1,115 @@
+        creator_request_id: CreatorRequestId,
+        name: Name,
+        tags: TagList = None,
+    ) -> CreateFirewallRuleGroupResponse:
+        region_details = Route53ResolverBackend.get()
+        id = get_route53_resolver_firewall_rule_group_id()
+        arn = aws_stack.get_route53_resolver_firewall_rule_group_arn(id)
+        firewall_rule_group: Optional[FirewallRuleGroup] = FirewallRuleGroup(
+            Id=id,
+            Arn=arn,
+            Name=name,
+            RuleCount=0,
+            Status="COMPLETE",
+            OwnerId=context.account_id,
+            ShareStatus="NOT_SHARED",
+            StatusMessage="Created Firewall Rule Group",
+            CreatorRequestId=creator_request_id,
+            CreationTime=datetime.now(timezone.utc).isoformat(),
+            ModificationTime=datetime.now(timezone.utc).isoformat(),
+        )
+        region_details.firewall_rule_groups[id] = firewall_rule_group
+        moto_route53resolver_backends[context.region].tagger.tag_resource(arn, tags or [])
+        return CreateFirewallRuleGroupResponse(FirewallRuleGroup=firewall_rule_group)
+
+    def delete_firewall_rule_group(
+        self, context: RequestContext, firewall_rule_group_id: ResourceId
+    ) -> DeleteFirewallRuleGroupResponse:
+        region_details = Route53ResolverBackend.get()
+        if not region_details.firewall_rule_groups.get(firewall_rule_group_id):
+            raise ResourceNotFoundException(
+                f"Can't find the resource with ID '{firewall_rule_group_id}'. Trace Id: '1-{get_random_hex(8)}-{get_random_hex(24)}'"
+            )
+        firewall_rule_group: Optional[FirewallRuleGroup] = region_details.firewall_rule_groups.pop(
+            firewall_rule_group_id
+        )
+        return DeleteFirewallRuleGroupResponse(FirewallRuleGroup=firewall_rule_group)
+
+    def get_firewall_rule_group(
+        self, context: RequestContext, firewall_rule_group_id: ResourceId
+    ) -> GetFirewallRuleGroupResponse:
+        region_details = Route53ResolverBackend.get()
+        firewall_rule_group: Optional[FirewallRuleGroup] = region_details.firewall_rule_groups.get(
+            firewall_rule_group_id
+        )
+        if not firewall_rule_group:
+            raise ResourceNotFoundException(
+                f"Can't find the resource with ID '{firewall_rule_group_id}'. Trace Id: '1-{get_random_hex(8)}-{get_random_hex(24)}'"
+            )
+        return GetFirewallRuleGroupResponse(FirewallRuleGroup=firewall_rule_group)
+
+    def list_firewall_rule_groups(
+        self, context: RequestContext, max_results: MaxResults = None, next_token: NextToken = None
+    ) -> ListFirewallRuleGroupsResponse:
+        region_details = Route53ResolverBackend.get()
+        firewall_group_rules: Optional[FirewallRuleGroupMetadataList] = []
+        for firewall_rule_group in region_details.firewall_rule_groups.values():
+            firewall_group_rules.append(
+                FirewallRuleGroupMetadata(
```

Comment:
We have a neat utility `select_from_typed_dict` in `utils.collections` to automatically select only the attributes defined in the model classes:
```
firewall_group_rules.append(select_from_typed_dict(FirewallRuleGroupMetadata, firewall_rule_group))
```

Reply:
Updated.